src/olympia/accounts/views.py

```diff
@@ -366,13 +370,32 @@ def delete(self, request, *args, **kwargs):
         return response
 
 
-class ProfileView(generics.RetrieveAPIView):
+class AccountViewSet(RetrieveModelMixin, GenericViewSet):
     authentication_classes = [JWTKeyAuthentication]
-    permission_classes = [IsAuthenticated]
-    serializer_class = UserProfileSerializer
-
-    def retrieve(self, request, *args, **kw):
-        return Response(self.get_serializer(request.user).data)
+    permission_classes = [AllowAny]
+    lookup_url_kwarg = 'user_id'
+    lookup_value_regex = r'\d+'
+    queryset = UserProfile.objects.all()
+    self_view = False
+
+    def get_serializer_class(self):
+        if (self.self_view or
+                acl.action_allowed_user(self.request.user,
+                                        amo.permissions.USERS_EDIT)):
+            return UserProfileSerializer
+        else:
+            return PublicUserProfileSerializer
+
+    def get_object(self):
+        lookup_value = self.kwargs.get(self.lookup_url_kwarg)
+        if self.request.user.is_authenticated():
```

Comment:
Proposed patch to make this more consistent with the rest of the APIs, fixing the `authentication_classes` as well: 
https://gist.github.com/diox/690f8391bf85db502fdfd3a0eff5110a

This makes the profile url a simple method on the viewset, to stop having to manually instantiate the viewset in `urls.py`